feature-libs/cart/base/root/facade/cart-guest-user.facade.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { facadeFactory } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { CART_BASE_CORE_FEATURE } from '../feature-name';
+import { CartGuestUser } from '../models';
+
+@Injectable({
+  providedIn: 'root',
+  useFactory: () =>
+    facadeFactory({
+      facade: CartGuestUserFacade,
+      feature: CART_BASE_CORE_FEATURE,
+      methods: ['createCartGuestUser', 'updateCartGuestUser'],
+    }),
+})
+export abstract class CartGuestUserFacade {
+  /**
+   * Creates guest user and assigns user to the cart.
+   */
+  abstract createCartGuestUser(
+    userId: string,
+    cartId: string,
+    guestUserDetails?: CartGuestUser
+  ): Observable<CartGuestUser>;
+
+  /**
+   * Updates guest user details.
+   */
+  abstract updateCartGuestUser(
+    userId: string,
+    cartId: string,
+    guestUserDetails: CartGuestUser
+  ): Observable<CartGuestUser>;
+}
```

Comment:
[NITPICK]
When the class name describes a domain (e.g. CartGuestUser), IMHO there is no need to repeat this domain name in each method. So e.g. `CartGuestUserService.create()` or `CartGuestUserService.update()` would be more concise and still correct.

As an example, see [UserProfileFacade](https://github.com/SAP/spartacus/blob/d0b209dae77cd503ed429e3dad1bb46ebfe0af19/feature-libs/user/profile/root/facade/user-profile.facade.ts#L23).
It's a general advice to make method names concise and easier to read.